sentinelsat/sentinel.py

```diff
@@ -527,6 +527,124 @@ def check_query_length(query):
         effective_length = len(query) + len(re.findall('[^-_.~0-9A-Za-z]', query))
         return effective_length / 3893
 
+    def _query_names(self, names):
+        """Find products by their names, e.g.
+        S1A_EW_GRDH_1SDH_20141003T003840_20141003T003920_002658_002F54_4DD1.
+
+        Note that duplicates exist on server, so multiple products can be returned for each name.
+
+        Parameters
+        ----------
+        names : list[string]
+            List of product names.
+
+        Returns
+        -------
+        dict[string, dict[str, dict]]
+            A dictionary mapping each name to a dictionary which contains the products with
+            that name (with ID as the key).
+        """
+
+        def chunks(l, n):
+            """Yield successive n-sized chunks from l."""
+            for i in range(0, len(l), n):
+                yield l[i:i + n]
+
+        products = {}
+        # 40 names per query fits reasonably well inside the query limit
+        for chunk in chunks(names, 40):
+            query = " OR ".join(chunk)
+            products.update(self.query_raw(query))
+
+        # Group the products
+        output = OrderedDict((name, dict()) for name in names)
+        for id, metadata in products.items():
+            name = metadata['identifier']
+            output[name][id] = metadata
+
+        return output
+
+    def check_existing(self, paths=None, ids=None, directory=None, delete=False):
+        """Verify the integrity of product files on disk.
+
+        The input can be a list of products to check or a list of IDs and a directory.
+
+        Integrity is checked by comparing the size and checksum of the file with the respective
+        values on the server.
+
+        In cases where multiple products with different IDs exist on the server for given product
```

Comment:
Good idea. That could be in [this line](https://github.com/sentinelsat/sentinelsat/pull/133/files/8d215f0c3adf41425b9132b1984781e443b9c029#diff-b41f2cfc13d00858923dd3e8da963187R626) - checking whether `product_infos` for any `name` is longer than 1. It would not need to contain much information on the products but direct the user to use the (planned) de-duplication method.